core/src/main/kotlin/org/evomaster/core/problem/rest/service/SecurityRest.kt

```diff
@@ -857,7 +868,179 @@ class SecurityRest {
+            )
+
+            if(successfulIndividuals.isEmpty()){
+                continue
+            }
 
+            // Take the smallest successful individual
+            val target = successfulIndividuals.minBy { it.individual.size() }
+
+            val actionIndex = RestIndividualSelectorUtils.findIndexOfAction(
+                target,
+                action.verb,
+                action.path,
+                statusGroup = StatusGroup.G_2xx
+            )
+
+            if(actionIndex < 0){
+                continue
+            }
+
+            // Slice to keep only up to the target action
+            val sliced = RestIndividualBuilder.sliceAllCallsInIndividualAfterAction(
+                target.individual,
+                actionIndex
+            )
+
+            // Try each XSS payload (but only add one test per endpoint)
+            for(payload in XSS_PAYLOADS){
+
+                // Create a copy of the individual
+                var copy = sliced.copy() as RestIndividual
+                val actionCopy = copy.seeMainExecutableActions().last() as RestCallAction
+
+                // Try to inject XSS payload into string fields
+                var payloadInjected = false
+
+                // Attempt to inject payload into parameters
+                actionCopy.parameters.filter { it is BodyParam || it is QueryParam || it is PathParam }.forEach { param ->
+
+                    // Skip if PathParam and payload contains "/" (would break URL structure)
+                    if(param is PathParam && payload.contains("/")){
+                        return@forEach
+                    }
+
+                    val genes = param.primaryGene().flatView()
+
+                    genes.forEach { gene ->
+                        if(gene !is StringGene) return@forEach
+
+                        try {
+                            // Check if constraints (min-max length) allow the payload
+                            if(payload.length >= gene.minLength && payload.length <= gene.maxLength){
+                                // Check if payload contains any invalid chars
+                                val hasInvalidChars = gene.invalidChars.any { payload.contains(it) }
+                                if(!hasInvalidChars){
+                                    // Set the XSS payload value
+                                    gene.value = payload
+                                    payloadInjected = true
+                                }
+                            }
+                        } catch(e: Exception){
+                            // Constraints might not allow the payload
+                            log.debug("Failed to inject XSS payload into ${gene.name}: ${e.message}")
+                        }
+                    }
+                }
+
+                if(!payloadInjected){
+                    continue
+                }
+
+
+                // Try to add a linked GET operation for stored XSS detection
```

Comment:
to avoid a too long and complex function here, refactor this block (ie the adding of GET) in a new private function